src/mame/sega/sega_beena.cpp

```diff
@@ -71,131 +102,2109 @@
+	sega_9h0_0008_state(const machine_config &mconfig, device_type type, const char *tag)
 		: driver_device(mconfig, type, tag)
-		, m_maincpu(*this, "arm7")
+		, m_maincpu(*this, "maincpu")
+		, m_workram(*this, "workram")
+		, m_pcm(*this, "pcm")
 		, m_cart(*this, "cartslot")
 		, m_cart_region(nullptr)
-		, m_bank(*this, "cartbank")
+		, m_screen_main(*this, "screen")
+		, m_gfxdecode(*this, "gfxdecode")
+		, m_palette(*this, "palette")
+		, m_paletteram(*this, "paletteram")
+		, m_tilemap_bg(*this, "tilemap_bg")
+		, m_tilemap_fg(*this, "tilemap_fg")
+		, m_tilemap_scroll_x(*this, "tilemap_scroll_x")
+		, m_tilemap_scroll_y(*this, "tilemap_scroll_y")
+		, m_tilemap_sprites(*this, "tilemap_sprites")
+		, m_bitmap(*this, "bitmap")
+		, m_io_sensor_regs(*this, "io_sensor_regs")
+		, m_io_auxiliary_regs(*this, "io_auxiliary_regs")
+		, m_io_cpu_config(*this, "CPU_CONFIG")
+		, m_io_video_config(*this, "VIDEO_CONFIG")
 	{ }
 
-	void sega_beena(machine_config &config);
+	void sega_9h0_0008(machine_config &config);
 
-private:
+protected:
+	virtual void device_post_load() override;
 	virtual void machine_start() override;
 	virtual void machine_reset() override;
+	virtual void video_start() override;
+
+	virtual void install_game_rom();
+	virtual void update_sensors(offs_t offset);
 
+	void beena_arm7_map(address_map &map);
+
+	void request_irq();
+	void request_fiq();
+
+	TIMER_DEVICE_CALLBACK_MEMBER(scanline);
+	virtual void update_crosshair(screen_device &screen);
+	void irq_wait_speedup();
 	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+	uint32_t video_reg_r(offs_t offset);
+	void video_reg_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	void pal_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint8_t tiles_layers_r(offs_t offset);
+	uint8_t tiles_sprites_r(offs_t offset);
+	void tiles_layers_w(offs_t offset, uint8_t data);
+	void tiles_sprites_w(offs_t offset, uint8_t data);
+	int32_t scroll_x(int32_t x, uint16_t i);
+	int32_t scroll_y(int32_t y, uint16_t i);
+	void draw_layer(bitmap_rgb32 &bitmap, const rectangle &cliprect, const uint16_t *tilemap, const uint8_t scroll_idx, const bool is_active, const bool is_overlay_rendered);
+	void draw_layer_tiles(bitmap_rgb32 &bitmap, const rectangle &cliprect, const uint16_t *tilemap, const uint8_t scroll_idx, const bool is_overlay_rendered);
+	void draw_layer_scanlines(bitmap_rgb32 &bitmap, const rectangle &cliprect, const uint16_t *tilemap, const uint8_t scroll_idx, const bool is_overlay_rendered);
+	void draw_sprites(bitmap_rgb32 &bitmap, const rectangle &cliprect, const bool is_overlay_rendered);
+	void draw_bitmap(bitmap_rgb32 &bitmap, const rectangle &cliprect);
+	void screen_blend(bitmap_rgb32 &bitmap, const rectangle &cliprect);
+	int32_t rescale_alpha_step(uint8_t step);
+	constexpr uint32_t alpha_blend_rgb_levels(uint32_t dst, uint32_t src, uint8_t level_b, uint8_t level_g, uint8_t level_r);
```

Comment:
Should this be `static`?  At a glance, it didn’t seem to reference any members.

Reply:
It can, I'll update it.